dateutil/relativedelta.py

```diff
@@ -592,6 +592,33 @@ def __repr__(self):
         return "{classname}({attrs})".format(classname=self.__class__.__name__,
                                              attrs=", ".join(l))
 
+    _msg = "'{}' not supported between instances of '{}' and '{}'"
+
+    if PY2:
+        def __lt__(self, other):
```

Comment:
I wonder if it's worth refactoring this as:

```python
if PY2:
    def __mk_cmp(op):
        def not_implemented(self, other):
            if isinstance(other, self.__class__):
                msg = "{} not supported between instances of {name} and {name}"
                raise TypeError(msg.format(op, self.__class__.__name__)
            return NotImplemented
        return not_implemented

...

class relativedelta(...):
    ...
    if PY2:
        # Backport Python 3-like failure semantics into this class
        __lt__ = __mk_cmp('<')
        __le__ = __mk_cmp('<=')
        __gt__ = __mk_cmp('>')
        __ge__ = __mk_cmp('>=')
```

That will cut down on the copy-paste.